src/link/Elf.zig

```diff
@@ -1956,7 +1956,7 @@ fn allocateTextBlock(self: *Elf, text_block: *TextBlock, new_block_size: u64, al
                 const sym = self.local_symbols.items[last.local_sym_index];
                 break :blk (sym.st_value + sym.st_size) - phdr.p_vaddr;
             } else 0;
-            const amt = try self.base.file.?.copyRangeAll(shdr.sh_offset, self.base.file.?, new_offset, text_size);
+            const amt = try self.base.file.?.copyRangeAll(shdr.sh_offset, self.base.file.?, new_offset, @intCast(usize, text_size));
```

Comment:
Ditto for `copyRangeAll`.

Reply:
I'm not sure if the solution I added is correct.

Reply:
You should fix `copyRange` to take a `u64` too as that's a file size.
How to deal with `os.copy_file_range` taking a `usize` for length is up to you, I'd add something like
```zig
const adjusted_len = math.cast(usize, len) catch math.maxInt(usize);
```

This is yet another example of how implementation details creep up into the application layer due to `os` being a layer taking the worst of each platform.

Reply:
> But that also means that it should return a u64 right?

`copyRange` has 1 caller, `copyRangeAll` has only 9 callsites (3 of them are in some test, the remaining in `Elf.zig` and `Coff.zig`).